src/Orchard.Cms.Web/Modules/Orchard.Roles/Controllers/AdminController.cs

```diff
@@ -72,7 +72,7 @@ IEnumerable<IPermissionProvider> permissionProviders
 
             var model = new RolesViewModel
             {
-                RoleEntries = roles.Select(BuildRoleEntry).ToList()
+                RoleEntries = roles.Select(r => BuildRoleEntry(r.Name)).ToList()
```

Comment:
So, do I change BuildRoleEntry method to expect a RoleNamesEntry type instead of string as parameter? 
Or do you prefer IRoleProvider provides another method that returns only names?
Of directly do we provide a method returning roles and each consumer will take what it needs?